src/main/java/discord4j/discordjson/json/PartialAttachmentData.java

```diff
@@ -0,0 +1,40 @@
+package discord4j.discordjson.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import discord4j.discordjson.possible.Possible;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutablePartialAttachmentData.class)
+@JsonDeserialize(as = ImmutablePartialAttachmentData.class)
+public interface PartialAttachmentData {
```

Comment:
The docs are not really clear about what is the Partial of attachment is.. if so maybe this class can be a "base" for the AttachmentData class.

Reply:
Looking at them, a partial attachment is like an attachment without the Id provided.
I was going to look for updating the signature in AttachmentData to Possible<Id> but that would have introduced a Breaking change.

I'll look for the previous comment details and think for another way to implement that.

Reply:
By looking deeper in Discord4J Rest implementation, it will be a better idea to handle those additional fields following the implementation from Discord4J Rest.

Both added fields by Discord (`payload_json`, `files[n]`, `attachments`) are client implementation based.